specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-10-01-preview/extensionTypes.json

```diff
@@ -0,0 +1,383 @@
+  "info": {
+    "version": "2021-10-01-preview",
+    "title": "ExtensionTypes",
+    "description": "Use these APIs to create extension resources through ARM, for Kubernetes Clusters."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterResourceName}/{clusterName}/providers/Microsoft.KubernetesConfiguration/extensionTypes/{extensionTypeName}": {
+      "get": {
+        "description": "Get Extension Type details",
+        "operationId": "ClusterExtensionType_Get",
+        "x-ms-examples": {
+          "ClusterExtensionType_Get_MaximumSet_Gen": {
+            "$ref": "./examples/ClusterExtensionType_Get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/2021-10-01-preview/parameters.json#/parameters/ClusterRpParameter"
+          },
+          {
+            "$ref": "../../common/2021-10-01-preview/parameters.json#/parameters/ClusterResourceNameParameter"
+          },
+          {
+            "$ref": "../../common/2021-10-01-preview/parameters.json#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ExtensionTypeNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ExtensionType"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterResourceName}/{clusterName}/providers/Microsoft.KubernetesConfiguration/extensionTypes": {
```

Comment:
>{clusterResourceName} [](http://example.com/codeflow?start=94&length=21)

This change breaks the ARM resource model, which requires alternating types and instances. This type of uri causes all kinds of problems. It's also a breaking change from your previous API version, since you are removing an existing endpoint.

There are other endpoints in this swagger with the same problem. You should fix your API to work properly with ARM.

Reply:
@mentat9 We are not removing an existing endpoint. The clusterRp combined with clusterResourceName should produce the following URIs:

- Microsoft.Kubernetes/connectedClusters
- Microsoft.ContainerService/managedClusters

We have this logic because we are a proxy resource and are used across multiple RPs and resourceTypes